pkg/proxy/proxy/transparent.go

```diff
@@ -108,13 +108,18 @@ func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate,
 	return TransparentProxyService{
 		Proxy:      p,
 		Ca:         ca,
-		NetworkLog: netlog.CaptureActivityLog(p, m),
+		NetworkLog: &netlog.NetworkActivityLog{},
 		Mode:       mode,
 		Policy:     pl,
 		mx:         m,
 	}
 }
 
+// LogActivity starts recording network activity in memory.
+func (t *TransparentProxyService) LogActivity() {
```

Comment:
Tbh I don't really like having a function like this so I'm open to changing it. My original thought was to add another constructor like
```
func NewTransparentProxyServiceWithLogging(p *goproxy.ProxyHttpServer, ca *tls.Certificate, mode PolicyMode, pl *policy.Policy) TransparentProxyService {}
```
which is essentially the same thing excepts it initializes the logger and the original constructor wouldn't. Wasn't sure if it was worth it to add another constructor though. wdyt?

Reply:
I just added a `TransparentProxyServiceOpts` struct for now. Not sure there's a use case for resetting the activity log currently so I'll maybe add that in the future. I've made NetworkLog private anyways since the mutex is also private so it doesn't make sense to create it outside of this class.

Reply:
hmm but without this method, there would be no way to start logging after creation (you'd need to name it "SkipLogging" instead). Given the current change, I think having a "RotateLog" method would fit nicely.

Reply:
I renamed it to SkipLogging and added a RotateLog function.